src/pages/all-posts.tsx

```diff
@@ -0,0 +1,154 @@
+import { Postcard } from '#ui/post-cards/big-post-card/big-post-card';
+import styled from 'styled-components';
+import { postCardsListMockArray } from '../mock-data/mock-data-posts';
+import { MediumPostcard } from '#ui/post-cards/medium-post-card/medium-post-card';
+import { ShortPostcard } from '#ui/post-cards/short-post-card/short-post-card';
+import { Link, Navigate, useParams } from 'react-router-dom';
+import { Title } from '#ui/title/title';
+import { ITab, MyTabPanel } from '#ui/tabs/tab-panel/tab-panel';
+import { useState } from 'react';
+import { PaginationMain } from '#features/pagination/pagination-main/pagination-main';
+
+export const AllListPosts: React.FC = () => {
+  const [selectedTab, setSelectedTab] = useState('all');
+
+  const BlogTabsMockArray: ITab[] = [
+    {
+      id: 'all',
+      title: 'All',
+    },
+    {
+      id: 'my-favourites',
+      title: 'My Favourites',
+    },
+    {
+      id: 'popular',
+      title: 'Popular',
+    },
+  ];
+
+  const postsAll = useParams();
+
+  if (!postsAll) {
```

Comment:
some mistake here - `postsAll` will be object in all cases; similar logic only necessary when your route has some params